include/DDML/GeometryInterface.h

```diff
@@ -20,20 +21,19 @@ namespace ddml {
  *  @date Mar 2023
  */
 
-class GeometryInterface {
-public:
-  virtual ~GeometryInterface() = default;
-
-  /** compute local direction in coordinate system that has the z-axis pointing
-   * into the calorimeter, normal to the layers
-   */
-  virtual G4ThreeVector localDirection(G4FastTrack const& aFastTrack) const = 0;
-
-  /** convert the local spacepoints to global spacepoints inside sensitive
-   * volumes
-   */
-  virtual void localToGlobal(G4FastTrack const& aFastTrack, std::vector<SpacePointVec>& spacepoints) const = 0;
-};
+template <typename T>
+concept GeometryInterface =
+    requires(const T t, const G4FastTrack& aFastTrack, std::vector<SpacePointVec>& spacepoints) {
+      /** compute local direction in coordinate system that has the z-axis pointing
+       * into the calorimeter, normal to the layers
+       */
+      { t.localDirection(aFastTrack) } -> std::same_as<G4ThreeVector>;
+
+      /** convert the local spacepoints to global spacepoints inside sensitive
+       * volumes
+       */
+      { t.localToGlobal(aFastTrack, spacepoints) } -> std::same_as<void>;
```

Comment:
The ones that are `void` should be harmonized. Some use `same_as<void>` others do not specify anything.